### PR TITLE
Add support for revocable credentials in vc_di handler

### PR DESCRIPTION
**Pull request by EmadAnwer**
This pull request supports issuing revocable verifiable credentials in the vc_di handler.

The main changes include:

- Adding revocation when creating a credential for proof. This includes creating a new revocation registry
- Fixing vc_di proof request issue by updating `vc_di/handler.py` **issue_credential** to include revocation missing data
- Adding integration tests to cover revocation of credentials.

This allows Aries to support the W3C Verifiable Credentials Data Model, including revocation of issued credentials. It enables revocable credential issuance and presentation via the vc_di protocols.

### CODE REVIEW

**Comment by ianco** [comment]
This line is incorrect:  https://github.com/hyperledger/aries-cloudagent-python/pull/2967/files#diff-f7d9c8d89f2442a02ac7a5924be4046e195d411791e4cbfd0fd9e7e2ecb987adL557

`if cred["proof"][0].get("rev_reg_id"):`

You can fetch these attributes from the credential, see for example:

https://github.com/ianco/aries-cloudagent-python/commit/1903eb6d45000ff57b65a2470bcf747adc8e5a39

(Not the most elegant approach but you can use it as an example)

**Review by ianco — CHANGES_REQUESTED** [review]
Hi @EmadAnwer I've doen a very quick review of the code and it looks ok, I need to review in a bit more detail and get one other dev to do a review.

In the meantime can you add a description to the pull request?  It will help anyone who needs to review, and also in the future if someone needs to look back to see what was done it will give them some context.  (Also release note re built based on the PR descriptions.)

Also there are a few code issues reported by one of the scanning tools, if you can take a look ...

I've built a couple of integration tests so I'll send those to you once I get them working :-S

**Comment by ianco** [comment]
FYI the SonarCloud errors (if you don't have access to the details) are:

```
Failed conditions
 [42.2% Coverage on New Code](https://sonarcloud.io/component_measures?id=hyperledger_aries-cloudagent-python&pullRequest=2967&metric=new_coverage&view=list) (required ≥ 80%)
 [24.7% Duplication on New Code](https://sonarcloud.io/component_measures?id=hyperledger_aries-cloudagent-python&pullRequest=2967&metric=new_duplicated_lines_density&view=list) (required ≤ 3%)
```

The coverage is just related to the unit tests.  The duplication is on the new `create_credential_w3c()` method which duplicates a lot of the existing `create_credential()` code (mostly the blocks around the revocation registry handling, which could be re-factored into helper methods).

**Comment by EmadAnwer** [comment]
Thanks for reviewing, im editing it and i will push asap

**Comment by EmadAnwer** [comment]
Thanks for reviewing, im editing it and i will push asap

**Comment by ianco** [comment]
@EmadAnwer  the coverage test is still failing on the `VCDICredFormatHandler` class (in `formats/vc_di/handler.py`).  

There are a couple of unit tests already that are marked as "skipped" (since they were added before revocation was implemented, I suspect just copied form the `indy` version of the unit test) - `async def test_issue_credential_revocable(self):` and `async def test_issue_credential_non_revocable(self):` - can you take a look at those and see what it would take to make these work with your updated revocation support?

Otherwise everything in the PR looks good to me.

**Comment by EmadAnwer** [comment]
> @EmadAnwer the coverage test is still failing on the `VCDICredFormatHandler` class (in `formats/vc_di/handler.py`).
> 
> There are a couple of unit tests already that are marked as "skipped" (since they were added before revocation was implemented, I suspect just copied form the `indy` version of the unit test) - `async def test_issue_credential_revocable(self):` and `async def test_issue_credential_non_revocable(self):` - can you take a look at those and see what it would take to make these work with your updated revocation support?
> 
> Otherwise everything in the PR looks good to me.


@ianco Thanks for the review and suggestions, sure I will take a look into it and update the PR

**Comment by ianco** [comment]
@EmadAnwer  it looks like the most recent commit (removing the dup code) is missing the DCO signoff

**Review by ianco — APPROVED** [review]

File: demo/features/0453-issue-credential.feature [hunk 1]
```diff
@@ -64,9 +64,10 @@ Feature: RFC 0453 Aries agent issue credential
 
     @WalletType_Askar_AnonCreds @SwitchCredTypeTest
```
Comment (jamshale) [inline demo/features/0453-issue-credential.feature:65]: Can we add an `@Release` tag here? So the tests run on our nightly and pre-release workflows.

**Review by jamshale — APPROVED** [review]
I would like to have this automatically integration tested with the `@Release` workflow. But, everything looks good :+1: 

Similar to work I'm doing right now, I want to make sure revocation registry rotation and other operations are working correctly in anoncreds. See https://github.com/hyperledger/aries-cloudagent-python/blob/main/aries_cloudagent/anoncreds/revocation.py#L875. This isn't in scope for this task. I will look at it separately.

**Comment by ianco** [comment]
@jamshale  I've merged the PR, I'll look at updating the tags on the integration tests later ...

**Pull request merged**